provider/pkg/provider/purrl/purrl.go

```diff
@@ -115,12 +116,12 @@ func callAPIEndpoint(method, url, body *string, responseCode *[]string, headers
 
 	resp, err := client.Do(request)
 	if err != nil {
-		return nil, fmt.Errorf("error calling API endpoint: %v", err)
+		return strPtr(string(resp.StatusCode)), nil, fmt.Errorf("error calling API endpoint: %v", err)
```

Comment:
Create a new function:

```go
func intPtr(i int) *int {
	return &i
}
```

and change to `return intPtr(resp.StatusCode), ...